src/main/java/com/trainigcenter/springtask/web/controller/GenreController.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/genres")
+public class GenreController {
+
+    private static final Logger logger = LogManager.getLogger(GenreController.class);
+
+    private final GenreService genreService;
+    private final ModelMapper modelMapper;
+
+    @Autowired
+    public GenreController(GenreService genreService, ModelMapper modelMapper) {
+        this.genreService = genreService;
+        this.modelMapper = modelMapper;
+    }
+
+    @GetMapping
+    public List<GenreDto> getAll() {
+        List<Genre> allGenres = genreService.getAll();
+        return allGenres.stream()
+                        .map(this::convertToDto)
+                        .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    public GenreDto getGenreById(@PathVariable("id") int id) {
+        Genre genre = Optional.ofNullable(genreService.getById(id))
+                              .orElseThrow(() -> new NotFoundException("Genre id:" + id + " not found"));
+
+        return convertToDto(genre);
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public GenreDto saveGenre(@Valid @RequestBody GenreDto genreDto) {
+        genreDto.setId(null);
+        genreDto.setName(mapGenreName(genreDto.getName()));
+
+        Genre genre = genreService.add(convertFromDto(genreDto));
+        return convertToDto(genre);
+    }
+
+    @PutMapping("/{id}")
+    @ResponseStatus(HttpStatus.CREATED)
+    public GenreDto updateGenre(@PathVariable("id") Integer id,
+                                @Valid @RequestBody GenreDto genreDto) {
+
+        Genre genre = convertFromDto(genreDto);
+        genre.setId(id);
+        genre.setName(mapGenreName(genre.getName()));
+
+        genre = Optional.ofNullable(genreService.update(genre))
+                        .orElseThrow(() -> new NotFoundException("Genre id:" + id + " not found"));
+
+        return convertToDto(genre);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteGenre(@PathVariable("id") Integer id) {
+
+        Genre genre = Optional.ofNullable(genreService.getByIdWithMovies(id))
+                              .orElseThrow(() -> new NotFoundException("Genre id:" + id + " not found"));
+
+        boolean isDeleted = genreService.delete(genre);
+
+        if (!isDeleted) {
+            throw new ForbiddenException(genre.getName() + " is take part in movies");
```

Comment:
You can say `Forbidden` if a user does not have access to some operation or resource. In you case, it is some sort of business constraint, not an authorization validation.